src/main/java/org/truffleruby/core/string/StringNodes.java

```diff
@@ -3240,25 +3241,50 @@ public Object stringToF(DynamicObject string, boolean strict,
         @Child private StringByteCharacterIndexNode byteIndexToCharIndexNode = StringNodesFactory.StringByteCharacterIndexNodeFactory.create(null);
         @Child private NormalizeIndexNode normalizeIndexNode = StringNodesFactory.NormalizeIndexNodeGen.create(null, null);
 
-        @Specialization(guards = { "isRubyString(pattern)", "isBrokenCodeRange(pattern)" })
-        public DynamicObject stringIndexBrokenCodeRange(DynamicObject string, DynamicObject pattern, int start) {
+        @Specialization(guards = { "isRubyString(pattern)", "isSingleByteSearch(string, pattern)" })
```

Comment:
Do we need `!isBrokenCodeRange(pattern)` ? Can a single-byte rope have a broken code range?

Reply:
Yes. You could have a rope marked as UTF-8, but which only consists of the first byte of a multi-byte character. E.g., `"ç".byteslice(0).valid_encoding? # false`

Reply:
Then this Specialization should ahve the guard otehrwise it returns inconsistent results with `stringIndexBrokenCodeRange`

Reply:
Please make the guards more explicit so it's clear they form a set of mutually-exclusive condition and cover the whole space, by adding `"!isBrokenCodeRange(pattern)"` here.